components/style/properties/properties.mako.rs

```diff
@@ -1623,23 +1646,9 @@ impl PropertyDeclaration {
             }
             _ => {}
         }
-        let longhand_id = match *self {
-            % for property in data.longhands:
-                PropertyDeclaration::${property.camel_case}(..) => {
-                    LonghandId::${property.camel_case}
-                }
-            % endfor
-            PropertyDeclaration::CSSWideKeyword(..) |
-            PropertyDeclaration::WithVariables(..) |
-            PropertyDeclaration::Custom(..) => {
-                debug_assert!(false, "unreachable");
-                // This value is never used, but having an expression of the same "shape"
-                // as for other variants helps the optimizer compile this `match` expression
-                // to a lookup table.
-                LonghandId::BackgroundColor
-            }
-        };
-        PropertyDeclarationId::Longhand(longhand_id)
+        PropertyDeclarationId::Longhand(
+            unsafe { *(self as *const _ as *const LonghandId) },
```

Comment:
Can we debug-assert this somehow at least, to ensure nobody messes up? I guess if they mess up it'd be completely orange but...

Also, please add a comment about why this is ok here so people don't scream :).